expression/builtin_test.go

```diff
@@ -27,6 +27,50 @@ import (
 	"github.com/pingcap/tidb/util/types"
 )
 
+type mockBuiltinFunc struct {
```

Comment:
Strictly speaking, this not a mock.

How about simply define a function that evals the `builtinFunc`.

```go
func evalBuiltinFunc(bf builtinFunc) (d types.Datum, err error)
```

Since the `row` argument is always nil, we don't need to add in `evalBuiltinFunc`.

Reply:
ok, but not all the `row` argument are nil, we still need to keep it.